Add `register_table` procedure support for iceberg table (#14375)
trinodb/trino: docs/src/main/sphinx/connector/iceberg.rst

    CALL iceberg.system.rollback_to_snapshot('testdb', 'customer_orders', 8954597067493422955)

> Register table

Reviewer: Place this content before https://trino.io/docs/current/connector/iceberg.html#migrating-existing-tables

Author: Addressed

----------------

> Iceberg supports creating table with existing metadata and data files.

Reviewer: `creating` is not appropriate.

The procedure _registers_ into the metastore an Iceberg table.

Author: Modified the text. Please review.

A SQL procedure ``system.register_table`` allows the caller to register a new table
with existing metadata and data::

>     CALL iceberg.system.register_table('testdb', 'customer_orders', 'hdfs://hadoop-master:9000/user/hive/warehouse/customer_orders-581fad8517934af6be1857a903559d44')

Reviewer: Use named parameters  in the example to provide a better explanation on how the procedure is being used.

Author: Addressed

In addition, you can provide a specific metadata file name to register table

> with specific metadata::

Reviewer: @mosabua  FYI

Table partitioning can also be changed and the connector can still
query data created before the partitioning change.

Register table
--------------
Iceberg and the Iceberg connector can register table into the metastore
with existing metadata and data files.

> A SQL procedure ``system.register_table`` allows the caller to register a new table

Author: Addressed!

Table partitioning can also be changed and the connector can still
query data created before the partitioning change.

Register table
--------------
The connector can register table into the metastore with existing metadata and data files.

An SQL procedure ``system.register_table`` allows the caller to register a new table

> with existing metadata and data files::

Author: Done

    CALL iceberg.system.register_table(schema_name => 'testdb', table_name => 'customer_orders', table_location => 'hdfs://hadoop-master:9000/user/hive/warehouse/customer_orders-581fad8517934af6be1857a903559d44')

In addition, you can provide a specific metadata file name to register table

> with specific metadata::

Author: Done

Table partitioning can also be changed and the connector can still
query data created before the partitioning change.

Register table
--------------
The connector can register table into the metastore with existing metadata and data files.

An SQL procedure ``system.register_table`` allows the caller to register an existing Iceberg
table in the metastore, using its existing metadata and data files::

    CALL iceberg.system.register_table(schema_name => 'testdb', table_name => 'customer_orders', table_location => 'hdfs://hadoop-master:9000/user/hive/warehouse/customer_orders-581fad8517934af6be1857a903559d44')

In addition, you can provide a specific metadata file name to register table
with specific metadata. This can be useful in case when you want to register
some specific table state, or the connector cannot automatically find out
the metadata to use::

    CALL iceberg.system.register_table(schema_name => 'testdb', table_name => 'customer_orders', table_location => 'hdfs://hadoop-master:9000/user/hive/warehouse/customer_orders-581fad8517934af6be1857a903559d44', metadata_file_name => '00003-409702ba-4735-4645-8f14-09537cc0b2c8.metadata.json')

> Due to security reasons, the procedure is enabled only when ``iceberg.allow-register-table-procedure`` is set to ``true``.

Author: Updated the doc.

Table partitioning can also be changed and the connector can still
query data created before the partitioning change.

Register table
--------------

> The connector can register table into the metastore with existing metadata and data files.

Reviewer 2: Iceberg calls them catalogs, since it's not necessarily a metastore.
```suggestion
The connector can register existing Iceberg tables with the catalog.
```

Author: Modified

An SQL procedure ``system.register_table`` allows the caller to register an existing Iceberg
table in the metastore, using its existing metadata and data files::

    CALL iceberg.system.register_table(schema_name => 'testdb', table_name => 'customer_orders', table_location => 'hdfs://hadoop-master:9000/user/hive/warehouse/customer_orders-581fad8517934af6be1857a903559d44')

In addition, you can provide a file name to register a table
with specific metadata. This may be used to register the table with
some specific table state, or may be necessary if the connector cannot

> automatically figure out the metadata to use::

Reviewer 2: ```suggestion
automatically figure out the metadata version to use::
```

Author: Modified
